### PR TITLE
Implement authorization on frontend

### PR DESCRIPTION
**Pull request by Marina0711**

### CODE REVIEW

File: client/src/components/HeaderComponent.tsx [hunk 1]
```diff
@@ -21,11 +23,21 @@ export const HeaderComponent = (props: HeaderComponentPropsType) => {
     const { title, isBackButton = false, isLogoutButton = false } = props;
     const navigation = useNavigation();
 
+    const goBack = () => {
+        navigation.goBack();
+    };
```
Comment (IslamRustam) [inline client/src/components/HeaderComponent.tsx:28]: why do we need a wrapper for goBack

File: client/src/helpers/AuthSubmitHelper.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        let user;
+        userStore.setStatus(StatusEnum.loading);
+
+        if (!values) {
+            user = await refreshToken();
+        } else if ((values as RegistrationFormType).phoneNumber) {
+            user = await registration(values as RegistrationFormType);
+        } else {
+            user = await login(values as LoginFormType);
+        }
```
Comment (IslamRustam) [inline client/src/helpers/AuthSubmitHelper.ts:21]: I don't think that it's good practice that this function depends on some fields of the argument. Can you think of a better approach?

Comment (IslamRustam) [reply client/src/helpers/AuthSubmitHelper.ts:21]: I am talking about this if else

File: client/src/components/HeaderComponent.tsx (hunk 1 above)
Comment (Marina0711) [reply client/src/components/HeaderComponent.tsx:28]: it seemed ugly to me to use goBack without a wrapper, and logout with a wrapper

**Pull request merged**